packages/forklift-console-plugin/src/modules/Providers/ProviderRow.tsx

```diff
@@ -30,76 +38,51 @@ interface CellProps {
   currentNamespace?: string;
 }
 
-/**
- * assumes that if condition is 'True' then
- * this is a positive state (success, "green")
- * i.e. ConnectionTestSucceeded
- */
-const toPositiveState = (conditionValue: string): 'Error' | 'Ok' | 'Unknown' => {
-  switch (conditionValue) {
-    case 'True':
-      return 'Ok';
-    case 'False':
-      return 'Error';
-    default:
-      return 'Unknown';
-  }
+const statusIcons = {
+  ValidationFailed: <RedExclamationCircleIcon />,
+  ConnectionFailed: <RedExclamationCircleIcon />,
+  Ready: <GreenCheckCircleIcon />,
+  Staging: <Spinner size="sm" />,
 };
 
-const fromProviderState = (status: ProviderStatus): 'Error' | 'Ok' | 'Unknown' | 'Loading' => {
-  switch (status) {
-    case 'Ready':
-      return 'Ok';
-    case 'ConnectionFailed':
-    case 'ValidationFailed':
-      return 'Error';
-    case 'Staging':
-      return 'Loading';
-    default:
-      return 'Unknown';
-  }
+const statusLabel = {
+  ValidationFailed: 'Validation Failed',
+  ConnectionFailed: 'Connection Failed',
+  Ready: 'Ready',
+  Staging: 'Staging',
 };
 
-const toNegativeState = (conditionValue: string): 'Error' | 'Ok' | 'Unknown' => {
-  switch (conditionValue) {
-    case 'True':
-      return 'Error';
-    case 'False':
-      return 'Ok';
-    default:
-      return 'Unknown';
+const StatusCell = ({ entity }: CellProps) => {
+  const allConditions = entity?.status?.conditions || [];
+  const phase = entity?.status?.phase;
+
+  if (allConditions.length === 0) {
+    return (
+      <>
+        {statusIcons[phase] || ''} {statusLabel[phase] || phase || ''}
+      </>
+    );
   }
-};
 
-const StatusCell = ({
-  value,
-  entity: { positiveConditions, negativeConditions, phase },
-  t,
-}: CellProps) => {
-  const allConditions = [
-    [positiveConditions, toPositiveState],
-    [negativeConditions, toNegativeState],
-  ].flatMap(
-    ([conditions, toState]: [
-      SupportedConditions,
-      (value: string) => 'Error' | 'Ok' | 'Unknown',
-    ]) => [
-      ...Object.values(conditions)
-        .filter(Boolean)
-        .map(({ message, status }) => {
-          return <StatusIcon key={message} status={toState(status)} label={message} />;
-        }),
-    ],
+  const conditionsTable = (
+    <Flex direction={{ default: 'column' }} spaceItems={{ default: 'spaceItemsXs' }}>
+      {allConditions.map((condition) => (
```

Comment:
Indeed removing the icons simplifies the logic as you don't need to distinguish between positive and negative conditions. But ...well ...
1. you loose icons:)
2. you cannot programmatically compare conditions